src/Base64Stream.php

```diff
@@ -134,18 +126,16 @@ public function eof()
      * Note that it's expected the underlying stream will only contain valid
      * base64 characters (normally the stream should be wrapped in a
      * PregReplaceFilterStream to filter out non-base64 characters).
-     *
-     * @param int $length
      */
-    private function fillBuffer($length)
+    private function fillBuffer(int $length) : void
     {
         $fill = 8192;
         while ($this->buffer->getSize() < $length) {
             $read = $this->stream->read($fill);
-            if ($read === false || $read === '') {
+            if ($read === '') {
```

Comment:
I'm worried about this change, was this an automatic change done by PhpCsFixer?  I feel like I had that for a reason... did you confirm that false isn't possible to be a return value from stream->read()?  iirc if the underlying stream's closed or something there's a possibility of false being returned, but that's just from the top of my head.

Reply:
PHPStan suggested it.  I looked into it and decided that read() could only return a string not a bool.  Double check my work, but I think it is correct.

Reply:
I do see that GuzzleHttp\Psr7\Stream::read does call fread, and that could return a false on error.

The better option would be to just use the == test, then false or empty string would pass.  It would make the code simpler too.

I will change to that if you want.

Reply:
I had a look and Stream::read() does check if 'read' returns false and throws a RuntimeException though.  I checked guzzlehttp/psr7 1.9.0 as well and that's the same there too.  I think we're good with the change, thank you :)

Reply:
Ah, I see now read() throws an exception if it get a false back from fread, so PHPStan was correct!